### ISSUE TITLE
avoid argument and column name confict

### ISSUE THREAD

**Issue by LittleShrimp1987** [mybatis/generator#620]
how about making column in dynamic support upper case like: 

public static final SqlColumn<String> USER_NAME = metric.userName; 

so we can avoid argument and column name confict. also the case column has the same name as table.

**Comment by a188037445** [comment]
I think these `SqlColumn` objects in `XXXDynamicSqlSupport` classes can be removed if there are no necessary reasons to retain them, we can use the members of the `SqlTable` objects, it's simple too.

**Comment by jeffgbutler** [comment]
The case where the column name is that same as the table name is already handled by the generator - we don't generate the short cut column in that case. I do understand that implementing this fix would resolve that in some cases.

The case where the column name and argument name are in conflict relates, I believe, to the generated *ByPrimaryKey methods. I don't think that appending an underscore as we do now is a terrible fix for that.

I think the thought here is that we should use the actual database column name as the field name. It is uppercase in many, but not all, databases. Most notably PostgreSQL stores mixed case identifiers.

The bottom line is that some name conflicts are inevitable and the generator as it is currently coded handles most of them in a consistent manner. So we not will make any changes for this.
